File: Java/0001-1000/001-100/51-60/53.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE, sum = 0;
        for (int i : nums) {
            if (sum < 0) {
                sum = 0;
            }
            sum += i;
            if (sum > max)
                max = sum;
        }
        return max;
    }
}

File: Python/0001-1000/001-100/51-60/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        m = max(nums)
        if m <= 0:
            return m
        s = 0
        for i in nums:
            s += i
            if s < 0:
                s = 0
            if s > m:
                m = s
        return m
